### PR TITLE
Allow blank values for optional CRD parameters

### PR DESCRIPTION
**Pull request by alxsap**

### CODE REVIEW

File: backend-k8s/steward-system/101-customResourceDefinition_PipelineRun.yaml [hunk 1]
```diff
@@ -64,25 +64,21 @@ spec:
                   pattern: '^[^\s]{1,}.*$'
                 repoAuthSecret: ###
                   type: string
-                  pattern: '^[^\s]{1,}.*$'
```
Comment (rinckm) [inline backend-k8s/steward-system/101-customResourceDefinition_PipelineRun.yaml]: I think we should not allow empty names for secrets.

Comment (alxsap) [reply backend-k8s/steward-system/101-customResourceDefinition_PipelineRun.yaml]: Same argument would apply for all other parameters where I had to remove the regex. JSON "", null and parameter not set at all, ends up in "" in the go struct.

Comment (anbrsap) [reply backend-k8s/steward-system/101-customResourceDefinition_PipelineRun.yaml]: For `spec.jenkinsFile.repoAuthSecret` the empty string would be ok. For `spec.secrets` and `spec.imagePullSecrets` we don't want empty or null list entries.

Comment (alxsap) [reply backend-k8s/steward-system/101-customResourceDefinition_PipelineRun.yaml]: Agree

**Pull request merged**